### PR TITLE
pkg add: libime

### PR DESCRIPTION
**Pull request by eatradish**
https://gitlab.com/fcitx/libime

### CODE REVIEW

File: extra-libs/libime/autobuild/defines [hunk 1]
```diff
@@ -0,0 +1,4 @@
+PKGNAME=libime
+PKGDES="This is a library to support generic input method implementation"
+PKGDEP="fcitx5 boost"
```
Comment (liushuyu) [inline extra-libs/libime/autobuild/defines]:
```suggestion
PKGDEP="fcitx-5 boost"
```

Okay, you forgot to change the name to match your changes to the name of `fcitx-5`

**Review by liushuyu — CHANGES_REQUESTED** [review]
Request changes

**Comment by RedL0tus** [comment]
Hmm... this library may belong to extra-i18n (just like librime)?

File: extra-libs/libime/autobuild/defines [hunk 2]
```diff
@@ -0,0 +1,4 @@
+PKGNAME=libime
+PKGDES="This is a library to support generic input method implementation"
```
Comment (RedL0tus) [inline extra-libs/libime/autobuild/defines]:
```suggestion
PKGSEC=libs
PKGDES="This is a library to support generic input method implementation"
```

Should have a PKGSEC.

**Comment by eatradish** [comment]
> Hmm... this library may belong to extra-i18n (just like librime)?

maybe...

**Review by RedL0tus — CHANGES_REQUESTED** [review]

File: extra-libs/libime/spec [hunk 3]
```diff
@@ -0,0 +1,2 @@
+VER=20180817
+GITSRC="https://gitlab.com/fcitx/libime"
```
Comment (RedL0tus) [inline extra-libs/libime/spec]:
```suggestion
GITSRC="https://gitlab.com/fcitx/libime"
GITCO="d612e0722a3fed5ede2138195c8b54d61dca4798"
```

Also GITCO here.

**Comment by liushuyu** [comment]
Add the following to `autobuild/prepare`

```bash
LIBTOPDIR="$(pwd)/build/src/libime"
for i in core pinyin table; do
export LD_LIBRARY_PATH="${LIBTOPDIR}/${i}":$LD_LIBRARY_PATH
done
```

File: extra-i18n/libime/autobuild/defines [hunk 4]
```diff
@@ -0,0 +1,4 @@
+PKGNAME=libime
+PKGDES="This is a library to support generic input method implementation"
+PKGDEP="fcitx-5 boost"
```
Comment (liushuyu) [inline extra-i18n/libime/autobuild/defines]: Okay, you need to change this thing back to `fcitx5` again

```suggestion
PKGDEP="fcitx5 boost"
```

File: extra-i18n/libime/autobuild/defines [hunk 5]
```diff
@@ -0,0 +1,4 @@
...
+BUILDEP="boost extra-cmake-modules git python-3"
```
Comment (liushuyu) [inline extra-i18n/libime/autobuild/defines]: For this part, dependencies that are already included in the `PKGDEP` don't need to be added again here.

```suggestion
BUILDEP="extra-cmake-modules git python-3"
```

**Review by RedL0tus — APPROVED** [review]

File: extra-i18n/libime/autobuild/defines [hunk 6]
```diff
@@ -0,0 +1,4 @@
+PKGNAME=libime
+PKGDES="This is a library to support generic input method implementation"
```
Comment (liushuyu) [inline extra-i18n/libime/autobuild/defines]:
```suggestion
PKGDES="A library to support generic input method implementation"
```

File: extra-i18n/libime/autobuild/defines [hunk 7]
```diff
@@ -0,0 +1,4 @@
+PKGNAME=libime
+PKGDES="A library to support generic input method implementation"
```
Comment (liushuyu) [inline extra-i18n/libime/autobuild/defines:2]:
```suggestion
PKGDES="A library to support generic input method implementation"
PKGSEC=libs
```

Comment (liushuyu) [reply extra-i18n/libime/autobuild/defines:2]: You need to add `PKGSEC`

**Pull request merged**